lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -240,6 +240,27 @@ class DefaultThumbnailsEngine(
         tracks.updateTracksInfo()
     }
 
+    override fun updateDataSources(dataSourcesNew: List<DataSource>) {
+        val currentVideoIds = dataSources.videoOrNull()?.map { it.mediaId() }
+        val newSourceIds = dataSourcesNew.map { it.mediaId() }.distinct()
```

Comment:
We can use distinctBy instead, so we don't have to go back and search for the same data sources by id again later